Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -644,103 +752,10 @@ void AODProducerWorkflowDPL::run(ProcessingContext& pc)
-        auto contributorsGID = recoData.getSingleDetectorRefs(trackIndex);
-        // extra info from sub-tracks
-        if (contributorsGID[GIndex::Source::ITS].isIndexSet()) {
-          isStoredITS[track.getRefITS()] = true;
-          const auto& itsOrig = recoData.getITSTrack(contributorsGID[GIndex::ITS]);
-          extraInfoHolder.itsClusterMap = itsOrig.getPattern();
-        }
-        if (contributorsGID[GIndex::Source::TPC].isIndexSet()) {
-          isStoredTPC[track.getRefTPC()] = true;
-          const auto& tpcOrig = recoData.getTPCTrack(contributorsGID[GIndex::TPC]);
-          extraInfoHolder.tpcChi2NCl = tpcOrig.getNClusters() ? tpcOrig.getChi2() / tpcOrig.getNClusters() : 0;
-          extraInfoHolder.tpcSignal = tpcOrig.getdEdx().dEdxTotTPC;
-          extraInfoHolder.tpcNClsFindable = tpcOrig.getNClusters();
-        }
-        addToTracksTable(tracksCursor, tracksCovCursor, track, -1, src);
-        addToTracksExtraTable(tracksExtraCursor, extraInfoHolder);
-      }
-      if (src == GIndex::Source::ITSTPCTOF && mFillTracksITSTPC) {
-        auto contributorsGID = recoData.getSingleDetectorRefs(trackIndex);
-        const auto& track = recoData.getITSTPCTOFTrack(contributorsGID[GIndex::Source::ITSTPCTOF]);
-        const auto& tofMatch = recoData.getTOFMatch(contributorsGID[GIndex::Source::ITSTPCTOF]);
-        extraInfoHolder.tofChi2 = tofMatch.getChi2();
-        const auto& tofInt = tofMatch.getLTIntegralOut();
-        extraInfoHolder.tofSignal = tofInt.getTOF(0); // fixme: what id should be used here?
-        extraInfoHolder.length = tofInt.getL();
-        // extra info from sub-tracks
-        if (contributorsGID[GIndex::Source::ITS].isIndexSet()) {
-          isStoredITS[track.getRefITS()] = true;
-          const auto& itsOrig = recoData.getITSTrack(contributorsGID[GIndex::ITS]);
-          extraInfoHolder.itsClusterMap = itsOrig.getPattern();
-        }
-        if (contributorsGID[GIndex::Source::TPC].isIndexSet()) {
-          isStoredTPC[track.getRefTPC()] = true;
-          const auto& tpcOrig = recoData.getTPCTrack(contributorsGID[GIndex::TPC]);
-          extraInfoHolder.tpcChi2NCl = tpcOrig.getNClusters() ? tpcOrig.getChi2() / tpcOrig.getNClusters() : 0;
-          extraInfoHolder.tpcSignal = tpcOrig.getdEdx().dEdxTotTPC;
-          extraInfoHolder.tpcNClsFindable = tpcOrig.getNClusters();
-        }
-        addToTracksTable(tracksCursor, tracksCovCursor, track, -1, src);
-        addToTracksExtraTable(tracksExtraCursor, extraInfoHolder);
-      }
-      if (src == GIndex::Source::MFT && mFillTracksMFT) {
-        const auto& track = tracksMFT[trackIndex.getIndex()];
-        isStoredMFT[trackIndex.getIndex()] = true;
-        addToMFTTracksTable(mftTracksCursor, track, -1);
-      }
-    }
-  }
+  // fixme: interaction time is undefined for unassigned tracks (?)
+  fillTrackTablesPerCollision(-1, -1, trackRefU, primVerGIs, recoData, tracksCursor, tracksCovCursor, tracksExtraCursor, mftTracksCursor);
```

Comment:
@nburmaso :  `trackRef` should be used here instead of `trackRefU`

Reply:
Hi @shahor02. Thank you for noticing. I have fixed the typo